### PR TITLE
deleting team prize

### LINKED ISSUE #719: Delete Team Prize
image

- If the status is winner and the entry is deleted, then we need to remove them from the prize as well
- update the correct prize entry with no id and no team name

### PR DESCRIPTION
**Pull request by twang0323**

### CODE REVIEW

File: src/app/api/teams/route.js [hunk 1]
```diff
@@ -72,20 +72,38 @@ export async function PUT(req) {
...
-          });
-        });
-        await deleteDoc(doc(db, "teams", object.uid));
-      } else if (attribute === "status") {
-        await updateDoc(doc(db, "teams", object.uid), {
-          status: status,
-        });
-      }
-    });
+    await Promise.all(
```
Comment (shahdivyank) [inline src/app/api/teams/route.js]: wait pause dont implement the promise.all yet, thats for a later time, keep the old structure but just add the if condition that u have below for it. also remove the comments

File: src/app/api/teams/route.js [hunk 2]
```diff
@@ -72,20 +70,36 @@ export async function PUT(req) {
   const { objects, attribute, status } = await req.json();
 
   try {
-    objects.forEach(async (object) => {
+    for (const object of objects) {
```
Comment (shahdivyank) [inline src/app/api/teams/route.js]: dont do for loop, keep it foreach

File: src/app/api/teams/route.js [hunk 3]
```diff
@@ -72,20 +70,36 @@ export async function PUT(req) {
...
             team: deleteField(),
           });
-        });
-        await deleteDoc(doc(db, "teams", object.uid));
+        }
+
+        const teamRef = doc(db, "teams", object.uid);
+
+        const teamSnapshot = await teamRef.get();
+        if (teamSnapshot.exists() && teamSnapshot.data().status === "winner") {
```
Comment (shahdivyank) [inline src/app/api/teams/route.js]: if we r deleting, then the team will always exist so no need to check

File: src/app/api/teams/route.js [hunk 4]
```diff
@@ -72,20 +70,36 @@ export async function PUT(req) {
...
           await updateDoc(doc(db, "users", member), {
             team: deleteField(),
           });
-        });
-        await deleteDoc(doc(db, "teams", object.uid));
+        }
+
+        const teamRef = doc(db, "teams", object.uid);
+
+        const teamSnapshot = await teamRef.get();
```
Comment (shahdivyank) [inline src/app/api/teams/route.js]: this get shouldnt work

**Comment by twang0323** [comment]
first change done

File: src/app/api/teams/route.js [hunk 5]
```diff
@@ -74,12 +72,28 @@ export async function PUT(req) {
   try {
     objects.forEach(async (object) => {
       if (attribute === "role") {
-        object.uids.forEach(async (member) => {
+        for (const member of object.uids) {
```
Comment (shahdivyank) [inline src/app/api/teams/route.js:75]: use a foreach loop

File: src/app/api/teams/route.js [hunk 6]
```diff
@@ -74,12 +72,28 @@ export async function PUT(req) {
...
           await updateDoc(doc(db, "users", member), {
             team: deleteField(),
           });
-        });
-        await deleteDoc(doc(db, "teams", object.uid));
+        }
+
+        const teamRef = doc(db, "teams", object.uid);
```
Comment (shahdivyank) [inline src/app/api/teams/route.js:81]: put the teamref directly into the delete statement

File: src/app/api/teams/route.js [hunk 7]
```diff
@@ -74,12 +72,28 @@ export async function PUT(req) {
...
+
+        const teamSnapshot = await getDocs(doc(db, "teams", object.uid));
+        if (teamSnapshot.data().status === "winner") {
+          const prizeSnapshot = await getDocs(collection(db, "prizes"));
```
Comment (shahdivyank) [inline src/app/api/teams/route.js:85]: this can be a query to prevent the foreach loop

**Pull request closed without merging**